Source/ChocolateyGui.Common.Windows/Services/ConfigFileWatcher.cs

```diff
@@ -0,0 +1,53 @@
+﻿// --------------------------------------------------------------------------------------------------------------------
+// <copyright company="Chocolatey" file="ConfigFileWatcher.cs">
+//   Copyright 2017 - Present Chocolatey Software, LLC
+//   Copyright 2014 - 2017 Rob Reynolds, the maintainers of Chocolatey, and RealDimensions Software, LLC
+// </copyright>
+// --------------------------------------------------------------------------------------------------------------------
+
+using System.IO;
+using Caliburn.Micro;
+using chocolatey.infrastructure.app.configuration;
+using chocolatey.infrastructure.services;
+using ChocolateyGui.Common.Models.Messages;
+using ChocolateyGui.Common.Services;
+using IFileSystem = chocolatey.infrastructure.filesystem.IFileSystem;
+
+namespace ChocolateyGui.Common.Windows.Services
+{
+    public class ConfigFileWatcher : IConfigFileWatcher
+    {
+        private readonly IFileSystem _fileSystem;
+        private readonly IEventAggregator _eventAggregator;
+        private readonly FileSystemWatcher _fileSystemWatcher;
+        private readonly IXmlService _xmlService;
+        private readonly string _configFile = chocolatey.infrastructure.app.ApplicationParameters.GlobalConfigFileLocation;

```

Comment:
This might be better as `_configFilePath`?

Reply:
Naming is hard, but yes that is a much better name.